### DISCUSSION TITLE
Should prisma offer alternative schema definition in a TS file

### DISCUSSION THREAD

**Discussion by capaj** [prisma/prisma#18217] [Q&A] [answered]
I know prisma is a multilanguage ORM basically, but 90 percen of people use it with typescript.
Compared to a DB like https://github.com/drizzle-team/drizzle-orm workflow in prisma always involves the codegen step whenever schema changes in  the schema file.
It would be interesting if Prisma added a TS schema support, because I believe that it would be possible to use Prisma in TS without codegen, whereas for other languages you could still use the `schema.prisma` file.

**Comment by nurul3101** [comment, marked as answer]
Just for reference, it's possible to automatically generate the PrismaClient when the schema file changes by using the `--watch` flag. Here are the [docs](https://www.prisma.io/docs/reference/api-reference/command-reference#continue-watching-the-schemaprisma-file-for-changes-to-automatically-re-generate-prisma-client) for it.

**Reply by ludralph** [reply]
Hey @capaj,

Do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!
